flang/lib/Semantics/resolve-names.cpp

```diff
@@ -1798,6 +1862,13 @@ void OmpVisitor::ProcessReductionSpecifier(
     Walk(clauses);
     PopScope();
   }
+
+  if (name) {
+    if (!symbol) {
+      symbol = &MakeSymbol(mangledName, Attrs{}, std::move(*reductionDetails));
+    }
+    name->symbol = symbol;
```

Comment:
It will be good to see a test for the multiple case as well. Does it override the symbol?

Reply:
I'm adding a new test for dupliocate symbols. There is one, but I think we need another different one.

And yes, we need to set the symbol on the original name, even if the actual name is not matching, so updating the symbol is required. If not, another portion finds the name and says it's not been resolved.